app/questionnaire/path_finder.py

```diff
@@ -59,19 +59,32 @@ def build_path(self):
                 group_blocks = self._build_blocks_for_group(group, instance_idx)
                 blocks += group_blocks
 
-            if group['id'] in first_groups:
-                this_location = Location(group['id'], 0, group['blocks'][0]['id'])
+            first_group_instance = self._get_first_group_instance_in_block(group, blocks)
+            if first_group_instance is not None:
+                if group['id'] in first_groups:
+                    this_location = Location(group['id'], first_group_instance, group['blocks'][0]['id'])
```

Comment:
If a block is skipped, do we want to be using the next group instance rather than the next block in the current group instance?

Reply:
I dont think so . If the skip condition points to an answer in a repeating group then the skip condition will need to be evaluated for each group instance, not for all instances within the block. If however it points to an answer not in a repeating group then it will skip or not skip the same for each group instance.

Reply:
Makes sense.  I guess I'm wondering about `group['blocks'][0]['id']`, might block 0 have been skipped?